### DISCUSSION TITLE
CRITICAL: Logon failed. Check 'USER' Password

### DISCUSSION THREAD

**Discussion by fedepasi** [SBFspot/SBFspot#590] [General]
Good morning,
I'm trying to use SBF on Home assistant with Raspberry 4.
I keep getting this error "CRITICAL: Logon failed. Check 'USER' Password" using the default password "0000".
With SMA Sunny Explorer I log in but I don't see any data, can you help me?
Thank you

**Comment by sbf-** [comment]
Well, it looks like the password is wrong. In SunnyExplorer, is there a lock near your inverter?

![image](https://user-images.githubusercontent.com/1931158/204131065-a87e2109-4a02-4a03-875e-8abf3b3306e4.png)

**Comment by Vitaminazeta** [comment]
Hello, I’m having the same problem too.
The ‘USER’ password is correct, but when I try to log in with Sunny Explorer, it asks me to change the default password.
I can log in and view some data, but in my case too it displays **_“there is no data currently available”_** and the closed padlock symbol.

The SMA display works fine with **user 0000**

When I try to log in with SBFspot, I get:
_CRITICAL: Logon failed. Check 'USER' Password_

Perhaps I need to use a password other than the default one to avoid the “change password” warning?

Has anyone found a way to resolve this LOG issue with SBFspot?
I’ve read that it’s possible to change the password, but you need the PUK codes. Mine is an old system that I found already installed.
Has anyone tried changing the password?

Thanks
Giacomo

**Comment by sbf-** [comment]
When you see the “there is no data currently available” message and the closed padlock you are NOT logged in.
So to me it's obvious, `0000` is a wrong password.

What do you mean with:
> The SMA display works fine with **user 0000**

**Comment by Vitaminazeta** [comment]
SMA display SunnyView it is a kind of small touch custom display delivered with the inverter, connecter via BT that shows the inverter status.
(It is 13 years old)
 
 Password is right for two reasons:

1. SunnyView can succesfully connect to the inverter via BT using USER password=0000
2. When I connect to the inverter with Sunny Explorer (Windows application) using USER password 0000 (or installer password 1111) I get a message "You must change default password". Not wrong password.

But SBFspot keep saying:   _CRITICAL: Logon failed. Check 'USER' Password_

The meaning of lock issue is that I don't have access rights to this specific device— ...the password you used for the general login doesn't match the password saved on this inverter, so Sunny Explorer can't read any configuration data from it.

I don't know what else to try, maybe SBFspot is non compatible with my inverter firmware

<img width="240" height="134" alt="lock" src="https://github.com/user-attachments/assets/f49a8e72-1e2a-4530-87eb-744b1ca93df6" />

Do you have the lock sign in your?

Giacomo

**Reply by sbf-** [reply]
FYI: I'm the developer of SBFspot, so I have some experience ;-)

You are using SE (Sunnyexplorer) version 2.x.x This version checks for user/0000 and installer/1111 default password combinations because it was unsafe. It forces you to change the password, but for some reason this doesn't work (on my recent inverter)

You can find an old version (1.07.29) [here](https://files.sma.de/downloads/SunnyExplorerSetup_v1_07_29_R.zip) - This version doesn't check for default passwords. But you will see you can't login either because of the wrong 0000 password.

SBFspot is compatible with your older inverter, it even works with my brand new hybrid inverter.

> SunnyView can succesfully connect to the inverter via BT using USER password=0000

Maybe your sunnyview device uses 0000 as bluetooth PIN to pair with your inverter. Besides `user/installer` account there is also a `service` account (hidden for customers)

> When I connect to the inverter with Sunny Explorer (Windows application) using USER password 0000 (or installer password 1111) I get a message "You must change default password". Not wrong password.

SE never says "Wrong password".
It just starts and shows the locked status

**Reply by Vitaminazeta** [reply]
I didn't realize you were the developer of SBFspot. :-)
Finally I managed to access my inverter with SBF!

With SunnyView, I forced the password to the same value, USER 0000. Once saved, SBFspot was able to access the inverter.

As I wrote in the other thread, I had simultaneously installed a clean version of SE v2.0, which allowed me to access the inverter without the lock symbol.

I don't know what action solved the problem.
Perhaps the password was incorrect, as you said, and forcing it from SView unlocked it.

I'd like to thank you for sharing SBFspot with the community. Thank you!
...and thanks for your support.

Ciao

Giacomo

**Comment by sbf-** [comment]
You can run SE in command line mode also. There you can see the failed login due to password:

Password OK
```
C:\Program Files (x86)\SMA\Sunny Explorer>SunnyExplorer.exe -userlevel user -password 0000 -exportdir c:\temp -plant "%USERPROFILE%\Documents\SMA\Sunny Explorer\My PV plant 1.sx2" -export energydaily -exportrange 20260801-20260822

C:\Program Files (x86)\SMA\Sunny Explorer>CSV-Export: establishing connection to plant ...
CSV-Export: plant login started ...
CSV-Export: current plant time is 22/08/2026 11:50:32
CSV-Export: starting export \nCSV-Export: completed
```

Password WRONG
```
C:\Program Files (x86)\SMA\Sunny Explorer>SunnyExplorer.exe -userlevel user -password 1111 -exportdir c:\temp -plant "%USERPROFILE%\Documents\SMA\Sunny Explorer\My PV plant 1.sx2" -export energydaily -exportrange 20260801-20260822

C:\Program Files (x86)\SMA\Sunny Explorer>CSV-Export: establishing connection to plant ...
CSV-Export: plant login started ...
CSV-Export: plant login failed!`
```

**Comment by Vitaminazeta** [comment]
Thanks for sending me both cases.

I finally managed to get SBFspot working. Unfortunately, I made two changes at the same time, and I'm not sure which one unlocked it.
First change: Uninstall SunnyExplorer from your PC by manually deleting all the old data files in the document folder. Install it from scratch and create a new connection.
Second change: There's an option on the SunnyView display to force the password. It forced the same password: 0000.




_SBFspot V3.9.12
Yet another tool to read power production of SMA solar inverters
(c) 2012-2025, SBF (https://github.com/SBFspot/SBFspot)
Compiled for Linux (LE) 64 bit with SQLite support
Commandline Args: -v -finq
Reading config '/usr/local/bin/sbfspot.3/SBFspot.cfg'
Sat Aug 22 14:01:34 2026: INFO: Starting...
sunrise: 06:26
sunset : 19:58
Connecting to 00:80:25:2D:10:6D (1/10)
Initialising...
SUSyID: 125 - SessionID: 918699476
SMA netID=02
Serial Nr: 7EF91D08 (2130255112)
BT Signal=7.8%
Logon OK
Local Host Time: 22/08/2026 14:01:46
Plant Time     : 22/08/2026 14:01:46 (0 sec)
TZ offset      : 3600 sec - DST: On
Last Time Set  : 22/08/2026 09:44:47
SUSyID: 138 - SN: 2130255112
Device Name:      SN: 2130255112
Device Class:     Solar Inverters
Device Type:      SB 4000TL-21
Software Version: 02.55.03.R
Packet status: 21
SUSyID: 138 - SN: 2130255112
Device Status:      Ok
SUSyID: 138 - SN: 2130255112
Device Temperature: 58.3°C
SUSyID: 138 - SN: 2130255112
GridRelay Status:      Closed
SUSyID: 138 - SN: 2130255112
Energy Production:
        EToday: 13.337kWh
        ETotal: 57255.410kWh
        Operation Time: 52664.01h
        Feed-In Time  : 50594.60h
SUSyID: 138 - SN: 2130255112
DC Spot Data:
        MPPT 1 Pdc:   1.431kW - Udc: 210.70V - Idc:  6.794A
        MPPT 2 Pdc:   1.191kW - Udc: 204.65V - Idc:  5.826A
        Calculated Total Pdc:   2.622kW
AC Spot Data:
        Phase 1 Pac :   2.532kW - Uac: 235.55V - Iac: 10.750A
        Phase 2 Pac :   0.000kW - Uac:   0.00V - Iac:  0.000A
        Phase 3 Pac :   0.000kW - Uac:   0.00V - Iac:  0.000A
        Total Pac   :   2.532kW - Calculated Pac:   2.532kW
        Efficiency  :   96.57%
SUSyID: 138 - SN: 2130255112
Grid Freq. : 50.00Hz
SUSyID: 138 - SN: 2130255112
Current Inverter Time: 22/08/2026 14:01:42
Inverter Wake-Up Time: 22/08/2026 06:22:55
Inverter Sleep Time  : 22/08/2026 14:01:45
* ArchiveDayData() *
startTime: 22/08/2026 00:00:00
* ArchiveMonthData() *
startTime: 01/08/2026 13:00:00
Reading events: 2026-Aug-01
Sat Aug 22 14:01:49 2026: INFO: Done._




Many thanks SBF for your help.

Giacomo
